src/style/style.js

```diff
@@ -322,7 +322,8 @@ class Style extends Evented {
             this._serializedLayers[layer.id] = layer.serialize();
             this._updateLayerCount(layer, true);
         }
-        this.dispatcher.broadcast('setProjection', this.map.transform.projectionOptions);
+
+        this._setProjection();
```

Comment:
This is the place were we likely want to set `this.stylesheet.projection` root property from the style if it's present, right?